.kokoro/tests/run_single_test.sh

```diff
@@ -41,6 +41,18 @@ fi
 nox -s "$RUN_TESTS_SESSION"
 EXIT=$?
 
+# Inject region tag data into the test log
+if [ "${INJECT_REGION_TAGS:-}" == "true" ]]; then
+    XUNIT_PATH="$PWD/sponge_log.xml"
+    XUNIT_TMP_PATH="tmp/drift_tmp.xml"
+    if [ -f "$XUNIT_PATH" ]; then
+        cp $XUNIT_PATH $XUNIT_TMP_PATH
+        cat "$XUNIT_TMP_PATH" | python3.7 "$PARSER_PATH" inject-snippet-mapping "$PWD" > "$XUNIT_PATH"
```

Comment:
If piped command `python3.7 "$PARSER_PATH" ...` fails, it will empty the `$XUNIT_PATH`.

Can you use a temporary file and move it after it succeeds?

Reply:
Done.